app/services/slug_setup.rb

```diff
@@ -0,0 +1,58 @@
+class SlugSetup
+    
+    attr_reader :name 
+
+    def self.setup(name)
+        new(name).send(:candidates)
+    end
+
+    def initialize(name)
+        @name = name
+    end
+    
+    def candidates
+        [   @name.method(:short_name), 
+            @name.method(:prolonged_name), 
+            @name.method(:orged), 
+            :name
+        ]
+    end
+    
+end
+
+class String
+    
+    NOT_WANTED = %w(the of for and in to)
+
+    def short_name
+        if self.match(/parochial church/i)
```

Comment:
usually we prefer 2 space indenting as per https://github.com/bbatsov/ruby-style-guide#spaces-indentation